kommand/changepassword.php

```diff
@@ -9,25 +8,19 @@
 <?php
 require ("dbfunctions.php");
 $dbh = db_connect();
-if ($subject_name) {
-    if (isvaliduser($dbh, $subject_name)) {
-        $info = get_items($dbh, "userid,email", "accounts", "username", $subject_name);
-        $subject_id = $info[0][0];
-        $email = $info[0][1];
-        if (!$password) {
-            srand(time());
-            $password = rand(0, 999999);
-        }
-        User::changePassword($subject_name,$password);
+if (isset($_POST['username'])) {
+    $user = User::get($_POST['username']);
+    if ($user) {
+        User::changePassword($user->username,$_POST['password'],null,true);
```

Comment:
Since the admins also need to flip the write-only bit, yes. Public plans have always required admin action to set up; this is not a new requirement.

I briefly considered allowing an account to set itself as write-only, but decided that'd just be a gigantic [footgun](https://en.wiktionary.org/wiki/footgun). Too many people will click every button available to them just to see what happens. (Or maybe that's just me... As part of testing the new write-only admin tool I did this to my own account, then immediately discovered I no longer had access to the tool and couldn't undo it without SQL access)

Reply:
Hahahaha. Yeah I think this strategy is fine. Needing admin intervention isn't enough of a burden to be worth changing it.